src/pages/VideoCall/permissions/CapabilitiesDialog.js

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect, useState } from 'react'
+import { Modal } from 'react-bootstrap'
+import { useTranslation } from 'react-i18next'
+import CameraMicrophonePermission from './CameraMicrophonePermission'
+import styled from 'styled-components'
+
+const CapabilitiesDialog = () => {
+    const { t } = useTranslation('videocall')
+    const [dialogDisplayed, setDialogDisplay] = useState(true)
+    const [checksPass, setCheckPass] = useState(false)
+    const [grantedMap, setGrantedMap] = useState({
+        cameraMicrophoneGranted: false,
+    })
+
+    const onCheckPass = (checkName) => {
+        if (checksPass) {
+            return
+        }
+        if (!grantedMap[checkName]) {
+            setGrantedMap({
+                ...grantedMap,
+                [checkName]: true,
+            })
+        }
+        if (Object.values(grantedMap).filter((value) => !!value)) {
+            setCheckPass(true)
+            setTimeout(() => {
+                setDialogDisplay(false)
+            }, 5000)
+        }
+    }
+
+    useEffect(() => {
+        window.iv = window.iv || {}
```

Comment:
why don't you define `window.iv = {}` directly?

Reply:
it's a good practice when touching global object linked to the window.

Reply:
You're right. I used to check if object exists. i take note of the tip